diff -aurN binutils-2.23.2/binutils/stabs.c binutils-2.23.2-fixed/binutils/stabs.c
--- binutils-2.23.2/binutils/stabs.c	2012-03-30 05:51:07 -0700
+++ binutils-2.23.2-fixed/binutils/stabs.c	2013-03-31 09:44:42 -0700
@@ -2593,7 +2593,7 @@
 	  bfd_boolean constp, volatilep, staticp;
 	  bfd_vma voffset;
 	  debug_type context;
-	  const char *physname;
+	  const char *physname = 0;
 	  bfd_boolean varargs;
 
 	  if (look_ahead_type != DEBUG_TYPE_NULL)
diff -aurN binutils-2.23.2/binutils/strings.c binutils-2.23.2-fixed/binutils/strings.c
--- binutils-2.23.2/binutils/strings.c	2012-02-08 20:51:44 -0800
+++ binutils-2.23.2-fixed/binutils/strings.c	2013-03-31 09:44:42 -0700
@@ -456,7 +456,7 @@
 {
   int c, i;
   long r = EOF;
-  unsigned char buf[4];
+  unsigned char buf[4] = { 0 };
 
   for (i = 0; i < encoding_bytes; i++)
     {
diff -aurN binutils-2.23.2/ld/ldfile.c binutils-2.23.2-fixed/ld/ldfile.c
--- binutils-2.23.2/ld/ldfile.c	2012-07-13 06:20:26 -0700
+++ binutils-2.23.2-fixed/ld/ldfile.c	2013-03-31 09:45:27 -0700
@@ -594,7 +594,7 @@
 ldfile_open_command_file_1 (const char *name, bfd_boolean default_only)
 {
   FILE *ldlex_input_stack;
-  bfd_boolean sysrooted;
+  bfd_boolean sysrooted = 0;
 
   ldlex_input_stack = ldfile_find_command_file (name, default_only, &sysrooted);
 
