diff -aurN binutils-2.37/libiberty/rust-demangle.c binutils-2.37-fixed/libiberty/rust-demangle.c
--- binutils-2.37/libiberty/rust-demangle.c	2021-07-15 07:19:46.000000000 -0700
+++ binutils-2.37-fixed/libiberty/rust-demangle.c	2021-09-06 14:02:58.616507700 -0700
@@ -75,10 +75,10 @@
   int version;
 
   /* Recursion depth.  */
-  uint recursion;
+  unsigned recursion;
   /* Maximum number of times demangle_path may be called recursively.  */
 #define RUST_MAX_RECURSION_COUNT  1024
-#define RUST_NO_RECURSION_LIMIT   ((uint) -1)
+#define RUST_NO_RECURSION_LIMIT   ((unsigned) -1)
 
   uint64_t bound_lifetime_depth;
 };
