diff -aurN boost_1_50_0/boost/thread/detail/config.hpp boost_1_50_0-fixed/boost/thread/detail/config.hpp
--- boost_1_50_0/boost/thread/detail/config.hpp	2012-06-16 14:10:08 -0700
+++ boost_1_50_0-fixed/boost/thread/detail/config.hpp	2012-07-05 00:24:17 -0700
@@ -150,19 +150,7 @@
 #elif defined(BOOST_THREAD_USE_DLL)   //Use dll
 #elif defined(BOOST_THREAD_USE_LIB)   //Use lib
 #else //Use default
-#   if defined(BOOST_THREAD_PLATFORM_WIN32)
-#       if defined(BOOST_MSVC) || defined(BOOST_INTEL_WIN)
-            //For compilers supporting auto-tss cleanup
-            //with Boost.Threads lib, use Boost.Threads lib
-#           define BOOST_THREAD_USE_LIB
-#       else
-            //For compilers not yet supporting auto-tss cleanup
-            //with Boost.Threads lib, use Boost.Threads dll
-#           define BOOST_THREAD_USE_DLL
-#       endif
-#   else
-#       define BOOST_THREAD_USE_LIB
-#   endif
+#   define BOOST_THREAD_USE_LIB
 #endif
 
 #if defined(BOOST_HAS_DECLSPEC)
