diff -aurN boost_1_49_0/libs/signals/src/named_slot_map.cpp boost_1_49_0-fixed/libs/signals/src/named_slot_map.cpp
--- boost_1_49_0/libs/signals/src/named_slot_map.cpp	2009-11-07 17:48:37 -0800
+++ boost_1_49_0-fixed/libs/signals/src/named_slot_map.cpp	2012-03-18 03:56:16 -0700
@@ -102,7 +102,7 @@
       i->first.disconnect();
       i = next;
     }
-    groups.erase(group);
+    groups.erase(static_cast<const_group_iterator>(group));
   }
 }
 
@@ -125,7 +125,7 @@
     }
 
     // Clear out empty groups
-    if (empty(g)) groups.erase(g++);
+    if (empty(g)) groups.erase(static_cast<const_group_iterator>(g++));
     else ++g;
   }
 }
