diff -aurN boost_1_54_0/boost/date_time/local_time/custom_time_zone.hpp boost_1_54_0-fixed/boost/date_time/local_time/custom_time_zone.hpp
--- boost_1_54_0/boost/date_time/local_time/custom_time_zone.hpp	2012-09-22 15:33:33 -0700
+++ boost_1_54_0-fixed/boost/date_time/local_time/custom_time_zone.hpp	2013-07-17 22:55:58 -0700
@@ -64,7 +64,7 @@
     //! True if zone uses daylight savings adjustments
     virtual bool has_dst() const
     {
-      return (dst_calc_rules_); //if calc_rule is set the tz has dst
+      return (bool) dst_calc_rules_; //if calc_rule is set the tz has dst
     }
     //! Local time that DST starts -- NADT if has_dst is false
     virtual posix_time::ptime dst_local_start_time(gregorian::greg_year y) const
diff -aurN boost_1_54_0/boost/log/utility/once_block.hpp boost_1_54_0-fixed/boost/log/utility/once_block.hpp
--- boost_1_54_0/boost/log/utility/once_block.hpp	2013-04-13 05:30:25 -0700
+++ boost_1_54_0-fixed/boost/log/utility/once_block.hpp	2013-07-17 22:56:02 -0700
@@ -176,7 +176,7 @@
  * been executed.
  */
 #define BOOST_LOG_ONCE_BLOCK_FLAG(flag_var)\
-    BOOST_LOG_ONCE_BLOCK_INTERNAL(\
+    BOOST_LOG_ONCE_BLOCK_FLAG_INTERNAL(\
         flag_var,\
         BOOST_LOG_UNIQUE_IDENTIFIER_NAME(_boost_log_once_block_sentry_))
 
diff -aurN boost_1_54_0/boost/thread/win32/thread_primitives.hpp boost_1_54_0-fixed/boost/thread/win32/thread_primitives.hpp
--- boost_1_54_0/boost/thread/win32/thread_primitives.hpp	2013-06-12 13:54:44 -0700
+++ boost_1_54_0-fixed/boost/thread/win32/thread_primitives.hpp	2013-07-17 22:56:06 -0700
@@ -19,7 +19,7 @@
 
 #ifndef BOOST_THREAD_WIN32_HAS_GET_TICK_COUNT_64
 #if _WIN32_WINNT >= 0x0600
-#define BOOST_THREAD_WIN32_HAS_GET_TICK_COUNT_64
+//#define BOOST_THREAD_WIN32_HAS_GET_TICK_COUNT_64
 #endif
 #endif
 
diff -aurN boost_1_54_0/libs/coroutine/build/Jamfile.v2 boost_1_54_0-fixed/libs/coroutine/build/Jamfile.v2
--- boost_1_54_0/libs/coroutine/build/Jamfile.v2	2013-05-22 09:10:52 -0700
+++ boost_1_54_0-fixed/libs/coroutine/build/Jamfile.v2	2013-07-17 22:55:54 -0700
@@ -40,7 +40,7 @@
     : detail/standard_stack_allocator_posix.cpp
     ;
 
-explicit yield_sources ;
+explicit allocator_sources ;
 
 lib boost_coroutine
     : allocator_sources
