diff -aurN freetype-2.13.2/CMakeLists.txt freetype-2.13.2-fixed/CMakeLists.txt
--- freetype-2.13.2/CMakeLists.txt	2023-08-25 11:12:52.000000000 -0700
+++ freetype-2.13.2-fixed/CMakeLists.txt	2024-01-07 12:32:03.686672800 -0800
@@ -297,7 +297,7 @@
 endif ()
 
 # Create the configuration file
-if (UNIX)
+if (UNIX AND NOT MSYS)
   check_include_file("unistd.h" HAVE_UNISTD_H)
   check_include_file("fcntl.h" HAVE_FCNTL_H)
 
@@ -423,7 +423,7 @@
   src/winfonts/winfnt.c
 )
 
-if (UNIX)
+if (UNIX AND NOT MSYS)
   list(APPEND BASE_SRCS "builds/unix/ftsystem.c")
 elseif (WIN32)
   list(APPEND BASE_SRCS "builds/windows/ftsystem.c")
