diff -aurN freetype-2.4.10/builds/unix/configure freetype-2.4.10-fixed/builds/unix/configure
--- freetype-2.4.10/builds/unix/configure	2012-06-14 21:46:55 -0700
+++ freetype-2.4.10-fixed/builds/unix/configure	2012-06-17 00:51:45 -0700
@@ -3231,7 +3231,7 @@
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-Os -fomit-frame-pointer"
   else
     CFLAGS="-g"
   fi
