diff -aurN gcc-4.6.3/libstdc++-v3/libsupc++/exception_ptr.h gcc-4.6.3-fixed/libstdc++-v3/libsupc++/exception_ptr.h
--- gcc-4.6.3/libstdc++-v3/libsupc++/exception_ptr.h	2011-03-31 18:07:46 -0700
+++ gcc-4.6.3-fixed/libstdc++-v3/libsupc++/exception_ptr.h	2012-03-18 04:14:02 -0700
@@ -97,7 +97,9 @@
       exception_ptr(exception_ptr&& __o) throw()
       : _M_exception_object(__o._M_exception_object)
       { __o._M_exception_object = 0; }
-#else
+#endif
+
+#if !defined (__GXX_EXPERIMENTAL_CXX0X__) || defined (_GLIBCXX_EH_PTR_COMPAT)
       typedef void (exception_ptr::*__safe_bool)();
 
       // For construction from nullptr or 0.
