diff -aurN gcc-4.8.0/gcc/c-family/c-common.c gcc-4.8.0-fixed/gcc/c-family/c-common.c
--- gcc-4.8.0/gcc/c-family/c-common.c	2013-03-08 23:54:02 -0800
+++ gcc-4.8.0-fixed/gcc/c-family/c-common.c	2013-03-31 00:01:56 -0700
@@ -238,9 +238,9 @@
 
 int flag_use_repository;
 
-/* The C++ dialect being used. C++98 is the default.  */
+/* The C++ dialect being used. C++11 is the default.  */
 
-enum cxx_dialect cxx_dialect = cxx98;
+enum cxx_dialect cxx_dialect = cxx11;
 
 /* Maximum template instantiation depth.  This limit exists to limit the
    time it takes to notice excessively recursive template instantiations.
diff -aurN gcc-4.8.0/gcc/c-family/c-opts.c gcc-4.8.0-fixed/gcc/c-family/c-opts.c
--- gcc-4.8.0/gcc/c-family/c-opts.c	2013-02-18 11:42:56 -0800
+++ gcc-4.8.0-fixed/gcc/c-family/c-opts.c	2013-03-31 00:01:56 -0700
@@ -210,7 +210,7 @@
   unsigned int i;
   struct cpp_callbacks *cb;
 
-  parse_in = cpp_create_reader (c_dialect_cxx () ? CLK_GNUCXX: CLK_GNUC89,
+  parse_in = cpp_create_reader (c_dialect_cxx () ? CLK_GNUCXX11 : CLK_GNUC89,
 				ident_hash, line_table);
   cb = cpp_get_callbacks (parse_in);
   cb->error = c_cpp_error;
