diff -aurN libjpeg-8d/include/jconfig.h libjpeg-8d-fixed/include/jconfig.h
--- libjpeg-8d/include/jconfig.h	2012-03-18 04:21:17 -0700
+++ libjpeg-8d-fixed/include/jconfig.h	2012-03-18 04:32:02 -0700
@@ -18,13 +18,6 @@
 /* Define this if you get warnings about undefined structures. */
 /* #undef INCOMPLETE_TYPES_BROKEN */
 
-/* Define "boolean" as unsigned char, not int, on Windows systems. */
-#ifdef _WIN32
-#ifndef __RPCNDR_H__		/* don't conflict if rpcndr.h already read */
-typedef unsigned char boolean;
-#endif
-#define HAVE_BOOLEAN		/* prevent jmorecfg.h from redefining it */
-#endif
 
 #ifdef JPEG_INTERNALS
 
diff -aurN libjpeg-8d/include/jmorecfg.h libjpeg-8d-fixed/include/jmorecfg.h
--- libjpeg-8d/include/jmorecfg.h	2011-08-31 04:35:12 -0700
+++ libjpeg-8d-fixed/include/jmorecfg.h	2012-03-18 04:32:02 -0700
@@ -150,24 +150,6 @@
 typedef unsigned int UINT16;
 #endif /* HAVE_UNSIGNED_SHORT */
 
-/* INT16 must hold at least the values -32768..32767. */
-
-#ifndef XMD_H			/* X11/xmd.h correctly defines INT16 */
-typedef short INT16;
-#endif
-
-/* INT32 must hold at least signed 32-bit values. */
-
-#ifndef XMD_H			/* X11/xmd.h correctly defines INT32 */
-#ifndef _BASETSD_H_		/* Microsoft defines it in basetsd.h */
-#ifndef _BASETSD_H		/* MinGW is slightly different */
-#ifndef QGLOBAL_H		/* Qt defines it in qglobal.h */
-typedef long INT32;
-#endif
-#endif
-#endif
-#endif
-
 /* Datatype used for image dimensions.  The JPEG standard only supports
  * images up to 64K*64K due to 16-bit fields in SOF markers.  Therefore
  * "unsigned int" is sufficient on all machines.  However, if you need to
@@ -225,22 +207,9 @@
 #endif
 
 
-/*
- * On a few systems, type boolean and/or its values FALSE, TRUE may appear
- * in standard header files.  Or you may have conflicts with application-
- * specific header files that you want to include together with these files.
- * Defining HAVE_BOOLEAN before including jpeglib.h should make it work.
- */
-
-#ifndef HAVE_BOOLEAN
-typedef int boolean;
-#endif
-#ifndef FALSE			/* in case these macros already exist */
-#define FALSE	0		/* values of boolean */
-#endif
-#ifndef TRUE
-#define TRUE	1
-#endif
+typedef unsigned char JPEG_boolean;
+#define JPEG_FALSE 0
+#define JPEG_TRUE  1
 
 
 /*
diff -aurN libjpeg-8d/include/jpeglib.h libjpeg-8d-fixed/include/jpeglib.h
--- libjpeg-8d/include/jpeglib.h	2011-08-27 09:15:42 -0700
+++ libjpeg-8d-fixed/include/jpeglib.h	2012-03-18 04:32:02 -0700
@@ -14,6 +14,8 @@
 #ifndef JPEGLIB_H
 #define JPEGLIB_H
 
+#include <stdio.h>
+
 /*
  * First we include the configuration files that record how this
  * installation of the JPEG library is set up.  jconfig.h can be
