diff -aurN make-4.0/job.c make-4.0-fixed/job.c
--- make-4.0/job.c	2013-10-05 16:12:24 -0700
+++ make-4.0-fixed/job.c	2013-11-25 14:29:11 -0800
@@ -194,7 +194,7 @@
 #if defined(WINDOWS32) && (__GNUC__ > 3 || _MSC_VER > 1300)
   /* %Id is only needed for 64-builds, which were not supported by
       older versions of Windows compilers.  */
-  sprintf (pidstring, "%Id", pid);
+  sprintf (pidstring, "%lld", pid);
 #else
   sprintf (pidstring, "%lu", (unsigned long) pid);
 #endif
diff -aurN make-4.0/main.c make-4.0-fixed/main.c
--- make-4.0/main.c	2013-10-08 21:22:40 -0700
+++ make-4.0-fixed/main.c	2013-11-25 14:27:39 -0800
@@ -809,7 +809,7 @@
       char hdl_string[2 * sizeof (sync_handle_t) + 2 + 1];
 
       /* Prepare the mutex handle string for our children.  */
-      sprintf (hdl_string, "0x%x", handle);
+      sprintf (hdl_string, "0x%llx", handle);
       sync_mutex = xmalloc (sizeof (struct stringlist));
       sync_mutex->list = xmalloc (sizeof (char *));
       sync_mutex->list[0] = xstrdup (hdl_string);
diff -aurN make-4.0/makeint.h make-4.0-fixed/makeint.h
--- make-4.0/makeint.h	2013-10-08 21:22:40 -0700
+++ make-4.0-fixed/makeint.h	2013-11-25 14:26:56 -0800
@@ -56,6 +56,8 @@
 #endif
 #include "gnumake.h"
 
+#define __USE_MINGW_ANSI_STDIO 1
+
 #ifdef  CRAY
 /* This must happen before #include <signal.h> so
    that the declaration therein is changed.  */
