diff -aurN make-4.1/main.c make-4.1-fixed/main.c
--- make-4.1/main.c	2014-10-05 09:24:51 -0700
+++ make-4.1-fixed/main.c	2015-04-26 15:20:33 -0700
@@ -3068,7 +3068,7 @@
 
         case flag:
         case flag_off:
-          if (!*(int *) cs->value_ptr == (cs->type == flag_off)
+          if ((!*(int *) cs->value_ptr) == (cs->type == flag_off)
               && (cs->default_value == 0
                   || *(int *) cs->value_ptr != *(int *) cs->default_value))
             ADD_FLAG (0, 0);
