diff -aurN pcre-8.32/pcre.h.generic pcre-8.32-fixed/pcre.h.generic
--- pcre-8.32/pcre.h.generic	2012-11-30 02:50:08 -0800
+++ pcre-8.32-fixed/pcre.h.generic	2012-12-08 19:39:33 -0800
@@ -46,25 +46,6 @@
 #define PCRE_PRERELEASE     
 #define PCRE_DATE           2012-11-30
 
-/* When an application links to a PCRE DLL in Windows, the symbols that are
-imported have to be identified as such. When building PCRE, the appropriate
-export setting is defined in pcre_internal.h, which includes this file. So we
-don't change existing definitions of PCRE_EXP_DECL and PCRECPP_EXP_DECL. */
-
-#if defined(_WIN32) && !defined(PCRE_STATIC)
-#  ifndef PCRE_EXP_DECL
-#    define PCRE_EXP_DECL  extern __declspec(dllimport)
-#  endif
-#  ifdef __cplusplus
-#    ifndef PCRECPP_EXP_DECL
-#      define PCRECPP_EXP_DECL  extern __declspec(dllimport)
-#    endif
-#    ifndef PCRECPP_EXP_DEFN
-#      define PCRECPP_EXP_DEFN  __declspec(dllimport)
-#    endif
-#  endif
-#endif
-
 /* By default, we use the standard "extern" declarations. */
 
 #ifndef PCRE_EXP_DECL
diff -aurN pcre-8.32/pcre.h.in pcre-8.32-fixed/pcre.h.in
--- pcre-8.32/pcre.h.in	2012-11-11 12:23:45 -0800
+++ pcre-8.32-fixed/pcre.h.in	2012-12-08 19:39:33 -0800
@@ -46,25 +46,6 @@
 #define PCRE_PRERELEASE     @PCRE_PRERELEASE@
 #define PCRE_DATE           @PCRE_DATE@
 
-/* When an application links to a PCRE DLL in Windows, the symbols that are
-imported have to be identified as such. When building PCRE, the appropriate
-export setting is defined in pcre_internal.h, which includes this file. So we
-don't change existing definitions of PCRE_EXP_DECL and PCRECPP_EXP_DECL. */
-
-#if defined(_WIN32) && !defined(PCRE_STATIC)
-#  ifndef PCRE_EXP_DECL
-#    define PCRE_EXP_DECL  extern __declspec(dllimport)
-#  endif
-#  ifdef __cplusplus
-#    ifndef PCRECPP_EXP_DECL
-#      define PCRECPP_EXP_DECL  extern __declspec(dllimport)
-#    endif
-#    ifndef PCRECPP_EXP_DEFN
-#      define PCRECPP_EXP_DEFN  __declspec(dllimport)
-#    endif
-#  endif
-#endif
-
 /* By default, we use the standard "extern" declarations. */
 
 #ifndef PCRE_EXP_DECL
diff -aurN pcre-8.32/pcreposix.h pcre-8.32-fixed/pcreposix.h
--- pcre-8.32/pcreposix.h	2011-12-28 08:57:51 -0800
+++ pcre-8.32-fixed/pcreposix.h	2012-12-08 19:39:34 -0800
@@ -109,16 +109,6 @@
   regoff_t rm_eo;
 } regmatch_t;
 
-/* When an application links to a PCRE DLL in Windows, the symbols that are
-imported have to be identified as such. When building PCRE, the appropriate
-export settings are needed, and are set in pcreposix.c before including this
-file. */
-
-#if defined(_WIN32) && !defined(PCRE_STATIC) && !defined(PCREPOSIX_EXP_DECL)
-#  define PCREPOSIX_EXP_DECL  extern __declspec(dllimport)
-#  define PCREPOSIX_EXP_DEFN  __declspec(dllimport)
-#endif
-
 /* By default, we use the standard "extern" declarations. */
 
 #ifndef PCREPOSIX_EXP_DECL
