diff -aurN SDL2-2.0.7/include/SDL_video.h SDL2-2.0.7-fixed/include/SDL_video.h
--- SDL2-2.0.7/include/SDL_video.h	2017-10-23 12:27:46.000000000 -0700
+++ SDL2-2.0.7-fixed/include/SDL_video.h	2018-02-18 18:38:31.169244200 -0800
@@ -39,6 +39,11 @@
 extern "C" {
 #endif
 
+extern int SDL_ClipCursorMarginLeft;
+extern int SDL_ClipCursorMarginTop;
+extern int SDL_ClipCursorMarginRight;
+extern int SDL_ClipCursorMarginBottom;
+
 /**
  *  \brief  The structure that defines a display mode
  *
diff -aurN SDL2-2.0.7/src/video/windows/SDL_windowswindow.c SDL2-2.0.7-fixed/src/video/windows/SDL_windowswindow.c
--- SDL2-2.0.7/src/video/windows/SDL_windowswindow.c	2017-10-23 12:27:47.000000000 -0700
+++ SDL2-2.0.7-fixed/src/video/windows/SDL_windowswindow.c	2018-02-18 18:38:31.201242000 -0800
@@ -142,6 +142,22 @@
     data->expected_resize = SDL_FALSE;
 }
 
+int SDL_ClipCursorMarginLeft   = 0;
+int SDL_ClipCursorMarginTop    = 0;
+int SDL_ClipCursorMarginRight  = 0;
+int SDL_ClipCursorMarginBottom = 0;
+
+void WIN_ClipCursor(const RECT * orig) {
+    RECT adjusted = *orig;
+
+    adjusted.left   += SDL_ClipCursorMarginLeft;
+    adjusted.top    += SDL_ClipCursorMarginTop;
+    adjusted.right  -= SDL_ClipCursorMarginRight;
+    adjusted.bottom -= SDL_ClipCursorMarginBottom;
+
+    ClipCursor(&adjusted);
+}
+
 static int
 SetupWindowData(_THIS, SDL_Window * window, HWND hwnd, HWND parent, SDL_bool created)
 {
@@ -261,7 +277,7 @@
             GetClientRect(hwnd, &rect);
             ClientToScreen(hwnd, (LPPOINT) & rect);
             ClientToScreen(hwnd, (LPPOINT) & rect + 1);
-            ClipCursor(&rect);
+            WIN_ClipCursor(&rect);
         }
     }
 
@@ -854,7 +870,7 @@
             if (GetClientRect(data->hwnd, &rect) && !IsRectEmpty(&rect)) {
                 ClientToScreen(data->hwnd, (LPPOINT) & rect);
                 ClientToScreen(data->hwnd, (LPPOINT) & rect + 1);
-                ClipCursor(&rect);
+                WIN_ClipCursor(&rect);
             }
         }
     } else {
