diff -aurN SDL2-2.0.1/include/SDL_video.h SDL2-2.0.1-fixed/include/SDL_video.h
--- SDL2-2.0.1/include/SDL_video.h	2013-10-23 21:05:29 -0700
+++ SDL2-2.0.1-fixed/include/SDL_video.h	2013-11-16 14:54:58 -0800
@@ -39,6 +39,11 @@
 extern "C" {
 #endif
 
+extern int SDL_ClipCursorMarginLeft;
+extern int SDL_ClipCursorMarginTop;
+extern int SDL_ClipCursorMarginRight;
+extern int SDL_ClipCursorMarginBottom;
+
 /**
  *  \brief  The structure that defines a display mode
  *
diff -aurN SDL2-2.0.1/src/video/windows/SDL_windowsevents.c SDL2-2.0.1-fixed/src/video/windows/SDL_windowsevents.c
--- SDL2-2.0.1/src/video/windows/SDL_windowsevents.c	2013-10-23 21:05:30 -0700
+++ SDL2-2.0.1-fixed/src/video/windows/SDL_windowsevents.c	2013-11-16 14:54:58 -0800
@@ -283,6 +283,8 @@
     return SDL_TRUE;
 }
 
+void WIN_ClipCursor(const RECT * orig);
+
 LRESULT CALLBACK
 WIN_WindowProc(HWND hwnd, UINT msg, WPARAM wParam, LPARAM lParam)
 {
@@ -657,7 +659,7 @@
             window_flags = SDL_GetWindowFlags(data->window);
             if ((window_flags & SDL_WINDOW_INPUT_GRABBED) &&
                 (window_flags & SDL_WINDOW_INPUT_FOCUS)) {
-                ClipCursor(&rect);
+                WIN_ClipCursor(&rect);
             }
 
             x = rect.left;
diff -aurN SDL2-2.0.1/src/video/windows/SDL_windowswindow.c SDL2-2.0.1-fixed/src/video/windows/SDL_windowswindow.c
--- SDL2-2.0.1/src/video/windows/SDL_windowswindow.c	2013-10-23 21:05:30 -0700
+++ SDL2-2.0.1-fixed/src/video/windows/SDL_windowswindow.c	2013-11-16 14:54:58 -0800
@@ -108,6 +108,22 @@
     SetWindowPos(hwnd, top, x, y, w, h, flags);
 }
 
+int SDL_ClipCursorMarginLeft   = 0;
+int SDL_ClipCursorMarginTop    = 0;
+int SDL_ClipCursorMarginRight  = 0;
+int SDL_ClipCursorMarginBottom = 0;
+
+void WIN_ClipCursor(const RECT * orig) {
+    RECT adjusted = *orig;
+
+    adjusted.left   += SDL_ClipCursorMarginLeft;
+    adjusted.top    += SDL_ClipCursorMarginTop;
+    adjusted.right  -= SDL_ClipCursorMarginRight;
+    adjusted.bottom -= SDL_ClipCursorMarginBottom;
+
+    ClipCursor(&adjusted);
+}
+
 static int
 SetupWindowData(_THIS, SDL_Window * window, HWND hwnd, SDL_bool created)
 {
@@ -219,7 +235,7 @@
             GetClientRect(hwnd, &rect);
             ClientToScreen(hwnd, (LPPOINT) & rect);
             ClientToScreen(hwnd, (LPPOINT) & rect + 1);
-            ClipCursor(&rect);
+            WIN_ClipCursor(&rect);
         }
     }
 
@@ -544,7 +560,7 @@
         GetClientRect(hwnd, &rect);
         ClientToScreen(hwnd, (LPPOINT) & rect);
         ClientToScreen(hwnd, (LPPOINT) & rect + 1);
-        ClipCursor(&rect);
+        WIN_ClipCursor(&rect);
     } else {
         ClipCursor(NULL);
     }
