diff -aurN SDL_mixer-1.2.12/configure SDL_mixer-1.2.12-fixed/configure
--- SDL_mixer-1.2.12/configure	2012-01-15 14:01:05 -0800
+++ SDL_mixer-1.2.12-fixed/configure	2012-03-19 23:22:35 -0700
@@ -13576,7 +13576,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lvorbisfile  $LIBS"
+LIBS="-lvorbisfile -lvorbis -logg $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
