diff -aurN sed-4.2.2/lib/msvc-inval.c sed-4.2.2-fixed/lib/msvc-inval.c
--- sed-4.2.2/lib/msvc-inval.c	2012-03-16 01:30:08 -0700
+++ sed-4.2.2-fixed/lib/msvc-inval.c	2013-08-07 00:50:31 -0700
@@ -30,7 +30,7 @@
 
 # if MSVC_INVALID_PARAMETER_HANDLING == DEFAULT_HANDLING
 
-static void cdecl
+static void __cdecl
 gl_msvc_invalid_parameter_handler (const wchar_t *expression,
                                    const wchar_t *function,
                                    const wchar_t *file,
@@ -47,7 +47,7 @@
 
 #  if defined _MSC_VER
 
-static void cdecl
+static void __cdecl
 gl_msvc_invalid_parameter_handler (const wchar_t *expression,
                                    const wchar_t *function,
                                    const wchar_t *file,
@@ -96,7 +96,7 @@
     }
 }
 
-static void cdecl
+static void __cdecl
 gl_msvc_invalid_parameter_handler (const wchar_t *expression,
                                    const wchar_t *function,
                                    const wchar_t *file,
diff -aurN sed-4.2.2/lib/regex_internal.h sed-4.2.2-fixed/lib/regex_internal.h
--- sed-4.2.2/lib/regex_internal.h	2012-12-22 05:21:52 -0800
+++ sed-4.2.2-fixed/lib/regex_internal.h	2013-08-07 00:49:41 -0700
@@ -448,7 +448,9 @@
 #define re_string_skip_bytes(pstr,idx) ((pstr)->cur_idx += (idx))
 #define re_string_set_index(pstr,idx) ((pstr)->cur_idx = (idx))
 
+#ifdef HAVE_ALLOCA_H
 #include <alloca.h>
+#endif
 
 #ifndef _LIBC
 # if HAVE_ALLOCA
diff -aurN sed-4.2.2/lib/wctype.in.h sed-4.2.2-fixed/lib/wctype.in.h
--- sed-4.2.2/lib/wctype.in.h	2012-12-22 05:21:53 -0800
+++ sed-4.2.2-fixed/lib/wctype.in.h	2013-08-07 00:49:41 -0700
@@ -40,6 +40,7 @@
    <wchar.h>.
    BSD/OS 4.0.1 has a bug: <stddef.h>, <stdio.h> and <time.h> must be
    included before <wchar.h>.  */
+# include <ctype.h>
 # include <stddef.h>
 # include <stdio.h>
 # include <time.h>
